src/test/ui/associated-consts/assoc-const.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(associated_const_equality)]
+
+pub trait Foo {
+  const N: usize;
+}
+
+pub struct Bar;
+
+impl Foo for Bar {
+  const N: usize = 3;
+}
+
+const TEST:usize = 3;
+
+
+fn foo<F: Foo<N=3>>() {}
+//~^ ERROR associated const equality is incomplete
```

Comment:
Why is the test still erroring?

Reply:
There was something in `astconv/mod.rs` which I did not yet fix, which adds predicates for ast type bindings currently, should I fix that in this PR?

Reply:
Ah no, that's fine. As long as you know why it's happening, I'm happy ^^